google-cloud-datastore/lib/google/cloud/datastore.rb

```diff
@@ -660,26 +660,16 @@ def self.new project_id: nil, credentials: nil, scope: nil, timeout: nil,
         client_config ||= configure.client_config
         emulator_host ||= configure.emulator_host
         if emulator_host
-          return Datastore::Dataset.new(
-            Datastore::Service.new(
-              project_id, :this_channel_is_insecure,
-              host: emulator_host, timeout: timeout,
-              client_config: client_config
-            )
-          )
+          return dataset_for_emulator(project_id, emulator_host, timeout,
```

Comment:
I understand why these methods were extracted, because of the Rubocop failures, but I don't want to add even more methods to the metaclass. I would rather see some creative formatting:

```ruby
return Datastore::Dataset.new(Datastore::Service.new(
  project_id, :this_channel_is_insecure,
  host: emulator_host, timeout: timeout,
  client_config: client_config
))
```

Or add exclusion comments around the method to disable the Rubocop rule:

```ruby
# rubocop:disable Metrics/MethodLength

def self.new project_id: nil, credentials: nil, scope: nil, timeout: nil,
             client_config: nil, emulator_host: nil, project: nil,
             keyfile: nil
  #...
end

# rubocop:enable Metrics/MethodLength
```

Reply:
Totally. I wasn't sure you'd be OK with disabling the rubocop metric. Fixed in cce5a7944f5a1ec626ce445591a7a84bed10b56d.